### ISSUE TITLE
[ws server]: doesn't reply with HTTP response on bad websocket HTTP upgrade request.  

### ISSUE THREAD

**Issue by 99Kies** [paritytech/jsonrpsee#818]
parachain version: `0.9.22`
start command: 
```bash
parachain-node --collator --chain ksm-mainnet.json --base-path /data \
 --port 3033  --ws-port 9944 --unsafe-ws-external --rpc-cors all \
 -- --execution wasm --chain kusama.json
```

The previous websocket supports http access.
![image](https://user-images.githubusercontent.com/45918704/177043923-cbdcec7c-8f09-4299-97e5-0a2cb6c91558.png)

After upgrading to 22, websocket no longer supports http access.
![image](https://user-images.githubusercontent.com/45918704/177044003-4cea0248-ba2c-4514-8908-3b492b9faf85.png)

**Comment by niklasad1** [comment]
Hey @99Kies 

Are you expecting a proper HTTP response to bad request on the HTTP request?
The request will fail to initiate the WebSocket handshake and it's just dropped on our side I think.

Should be quite easy to fix though, are you using this to find out when the websocket server has been started?

**Comment by 99Kies** [comment]
> Hey @99Kies
> 
> Are you expecting a proper HTTP response to the HTTP request? The request will fail to initiate the WebSocket handshake and it's just dropped on our side I think.
> 
> Should be quite easy to fix though, are you using this to find out when the websocket server has been started?

We will try to check if the service is healthy via the HTTP protocol. I think this is also necessary.

**Comment by niklasad1** [comment]
It's bit "expensive" to open a connection for that everytime.

Recently, we added `server-side` pings to WS Server which is enabled in polkadot v0.9.25. It will send out a pings periodically every 30s and it might be easier for you to just listen on those instead of opening a new connection all the time.

However, we should fix it anyway.

**Comment by tinydjp** [comment]
Hi @niklasad1 as we're using AWS to run our nodes, it seems that only ALB can support the wss protocol. In that case, ALB can only use HTTP(S) to run health check. We tried to use Network Load Balancer, the health check succeeded with TCP, but failed to connect via ws/wss.

**Comment by niklasad1** [comment]
Alright, I see.

This is fixed by https://github.com/paritytech/jsonrpsee/pull/819 anyway

**Comment by niklasad1** [comment]
Closed by #819
